sky/cli.py

```diff
@@ -162,7 +162,7 @@ def _get_cluster_records_and_set_ssh_config(
                     # updating skypilot.
                     f'\'{sys.executable} {sky.__root_dir__}/templates/'
                     f'websocket_proxy.py '
-                    f'{server_common.get_server_url().split("://")[1]} '
+                    f'{server_common.get_server_url().replace("http://", "ws://").replace("https://", "wss://").split("://")[1]} '
```

Comment:
it seems after splitting, the URL passed to the executable will not be affected by the replacement. Should we remove the split, and make the replacement in `sky/templates/websocket_proxy.py` instead?

Reply:
I see what you mean, I left the split there, it shouldn't be there! What the best way to set this up to test locally on MacOS? I tried building the Dockerfile, but it just pulls the latest `sky` cli nightly with pip...

My thinking was to make this backwards-compatible, so that a user who still has a "non-protocoled" hostname in their config can still use `sky/templates/websocket_proxy.py` as-is (_there's a check for `://` in the hostname_), but that any new configs created after this PR is merged will have either `http://` or `https://` in the config, which will better inform `sky/templates/websocket_proxy.py` re which websockets prefix to use.

Reply:
For testing it, a simple test might be mock the `is_api_server_local` to be `False` all the time? https://github.com/skypilot-org/skypilot/blob/bc5394a2ac897a5365ca5698cfe8dae40f33255b/sky/server/common.py#L95-L97

Reply:
What I'm after in terms of advice on testing is more about how to build the CLI locally? I was hoping to do it in a dockerfile to keep dependencies etc clean, but the Dockerfile seems to just pull the latest nightly with pip. Is there a "best practice" way to test this locally?